client/src/screens/AddItemScreen/AddAndUpdateItemScreen.tsx

```diff
@@ -0,0 +1,193 @@
+
+export const AddAndUpdateItemScreen = ({navigation, route}: AddItemScreenProps) => {
+    const dispatch = useAppDispatch();
+    const {_id, name} = route.params.category;
+    const {itemName, imageUri, itemId} = route.params;
+    const category = name;
+    const categoryId = _id;
+    const [newItemName, setNewItemName] = useState<string>('');
+    const [editItemName, setEditItemName] = useState<string>('');
+    const [searchItem, setSearchItem] = useState<string>('');
+    const [image, setImage] = useState<string>(imageUri || '');
+    const [imagesList, setImagesList] = useState<string[]>([]);
+    const [loading, setLoading] = useState(false);
+    const email: string | undefined = useAppSelector(selectEmail);
+    const needToSearch = itemId ? editItemName === '' && editItemName === newItemName && image === '' || image === imageUri && editItemName === newItemName : image === '';
+    const handleItemNameChanged = (e: NativeSyntheticEvent<TextInputChangeEventData>) => {
+        itemId ? setEditItemName(e.nativeEvent.text) : setNewItemName(e.nativeEvent.text);
+    }
+
+    const handleSearchItemChanged = (e: NativeSyntheticEvent<TextInputChangeEventData>) => {
+        setSearchItem(e.nativeEvent.text);
+    }
+
+    const handleClearChanges = () => {
+        setEditItemName('');
+        setSearchItem('');
+        setImage(imageUri || '')
+    }
+
+    const handleOnPress = () => {
+        setImage('');
+        setImagesList([]);
+        return axiosInstance.get(`/photos/${searchItem}`)
+            .then((response) => setImagesList(response.data))
+            .catch(console.error);
+    };
+
+    const uploadImageToS3 = async () => {
+        const imageType = image.split('.')[1];
+        const imageName = `s3-${email}-${Date.now()}.${imageType}`;
+        const response = await axiosInstance.post('/photos/upload/', {imageName});
+        const img = await fetch(image);
+        const blob = await img.blob();
+        await fetch(response.data.url, {
+            method: "PUT",
+            body: blob,
+        });
+        return imageName;
+    }
+
+    const handleSave = async () => {
+        if (!image) {
+            alert('בבקשה בחר תמונה');
+            return;
+        }
+        setLoading(true);
+
+        try {
+            const imageUrl = image.startsWith('file') ? await uploadImageToS3() : image;
+            if (itemId) {
+                await editItem(itemId, categoryId, imageUrl, editItemName).catch(console.log);
+            } else {
+                await axiosInstance.post('/items/', {name: newItemName, imageUrl, categoryId});
+            }
+        } catch (error) {
+            setLoading(false)
+            Toast.show({
+                type: 'error',
+                text1: 'השמירה נכשלה',
+                text2: 'הפרטים שהזנת לא נשמרו במערכת ⛔️',
+            });
+            console.log(error);
+        }
+
+        setLoading(false);
+
+        const fetchData = async () => {
```

Comment:
NOOOOOOO אני בשוק שלא שמתי לב לזה עד עכשיו
By doing this, in case of an error it'll show the error toast, then it'll show the success toast
Move fetchData + Toast.show('success') + (maybe navigation.pop too?) inside the `try catch`.
The `setLoading(false)` can remain at the end.

Reply:
סידרתי את זה עם return